Sources/SwiftParser/Expressions.swift

```diff
@@ -1271,20 +1271,18 @@ extension Parser {
     flavor: ExprFlavor
   ) -> RawMacroExpansionExprSyntax {
     var (unexpectedBeforePound, pound) = self.expect(.pound)
-    if pound.trailingTriviaByteLength != 0 {
+    if pound.trailingTriviaByteLength > 0 || currentToken.leadingTriviaByteLength > 0 {
       // If there are whitespaces after '#' diagnose.
-      unexpectedBeforePound = RawUnexpectedNodesSyntax(combining: unexpectedBeforePound, pound, arena: self.arena)
-      pound = self.missingToken(.pound)
+      let diagnostic = TokenDiagnostic(
+        .extraneousTrailingWhitespaceError,
+        byteOffset: pound.leadingTriviaByteLength + pound.tokenText.count
+      )
+      pound = pound.tokenView.withTokenDiagnostic(tokenDiagnostic: diagnostic, arena: self.arena)
```

Comment:
Because the unexpected route didn’t have any way of representing warnings, just errors. If there are unexpected or missing tokens in the syntax tree, we always consider the tree as having an error (which makes sense to me) and `TokenDiagnostic` is the only way we can currently communicate warnings.